config/common.mk

```diff
@@ -16,7 +16,7 @@ endif
 # Custom packages
 
 ifdef EXTENDROM_PACKAGES
-PRODUCT_PACKAGES += $(shell echo $$EXTENDROM_PACKAGES)
+PRODUCT_PACKAGES += $(shell echo $$EXTENDROM_PACKAGES | sed "s/Magisk//g")
```

Comment:
this would catch only a subset of supported EXTENDROM_PACKAGES values (I had mentioned that one as a workaround for your specific setup only)

we must catch all [these](https://github.com/sfX-android/android_vendor_extendrom/blob/a834a84b8b7c81a4f5e1bbea58e5e6948d3d10fb/repo/packages.txt#L31-L44) so that line is not enough.

If I am not mistaken this might work:

`PRODUCT_PACKAGES += $(shell echo "$$EXTENDROM_PACKAGES" | tr ' ' '\n' | sed "s/$$MAGNAME//g")`

If that works it has the benefit that we can re-use [$MAGNAME](https://github.com/sfX-android/android_vendor_extendrom/blob/a834a84b8b7c81a4f5e1bbea58e5e6948d3d10fb/er.sh#L440C5-L440C12) so if smth changes there no further adjustments are needed here.

If it does **not** work this might work though:

```
MAGNAME := $(shell echo "$$EXTENDROM_PACKAGES" | tr ' ' '\n' | grep -E '^Magisk$|Magisk_v[0-9]+\.[0-9]+$|SignMagisk$')

PRODUCT_PACKAGES += $(shell echo "$$EXTENDROM_PACKAGES" | tr ' ' '\n' | sed "s/$(MAGNAME)//g")

# if that gives syntax errors etc try:
PRODUCT_PACKAGES += $(shell echo "$$EXTENDROM_PACKAGES" | tr ' ' '\n' | sed "s/$MAGNAME//g")
```

as said the first one would be much better though and I am not sure about the syntax of the second so you might get errors here. could you test that approach?